core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -588,7 +588,10 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
             if (doRebalance)
               syncedRebalance
           } catch {
-            case t: Throwable => error("error during syncedRebalance", t)
+            case t: Throwable => {
+              error("error during syncedRebalance", t)
+              shutdown()
```

Comment:
the problem is your not guaranteed when the next rebalance will happen, if at all.
some here's a non-hypothetical scenario - a consumer has failed to rebalance, but remains "alive" (its still registered in ZK). the rest of the consumers succeed to rebalance and leave a chunk of partitions to the failed consumer (because its still alive on ZK).

and you end up with orphaned partitions for an unbounded amount of time.